### PR TITLE
#5 - coupler_flux_ocean_to_ice and coupler_flux_ice_to_ocean

### PR DESCRIPTION
**Pull request by mlee03**
In this PR,

`coupler_flux_ocean_to_ice` and `coupler_flux_ice_to_ocean` have been defined and have been used to replace sections of coupler_main that calls _flux_ocean_to_ice_ and _flux_ice_to_ocean_ subroutines .

Subroutine `coupler_flux_ice_to_ocean` is called twice in _coupler_main_ and contains the option to set _slow_ice_ocean_pelist_ as the current pelist for the following reason:   at the top of the outer loop, current pelist is set to _slow_ice_ocean_pelist_ in `coupler_flux_ocean_to_ice`.  Thus, the pelist is already set to the correct one when `coupler_flux_ice_to_ocean` is called immediately afterwords.  However, the current pelist is not set to _slow_ice_ocean_pelist_ when `coupler_flux_ice_to_ocean` is called at the bottom of the outer loop.   The optional argument is not specified in the first call and is set in the second call.

Pending:  all the "if(pe)" statements remain unchanged in _coupler_main_ in case MPI parallelization details benefit the users.

### CODE REVIEW

File: full/coupler_main.F90 [hunk 1]
```diff
@@ -371,7 +371,7 @@ program coupler_main
   character(len=32) :: timestamp
 
   type(coupler_clock_type) :: coupler_clocks
-
```
Comment (mlee03) [inline full/coupler_main.F90:374]: I have no idea what's going on with these trailing spaces

File: full/full_coupler_mod.F90 [hunk 2]
```diff
@@ -1694,19 +1693,80 @@ subroutine coupler_flux_check_stocks(nc, Time, Atm, Land, Ice, Ocean_state, coup
...
       call fms_mpp_set_current_pelist()
       call flux_check_stocks(Time=Time, Atm=Atm, Lnd=Land, Ice=Ice, Ocn_state=Ocean_state)
     endif
     call fms_mpp_clock_end(coupler_clocks%flux_check_stocks)
-    
+
   end subroutine coupler_flux_check_stocks
-  
+
+  !> \brief This subroutine calls flux_ocean_to_ice
```
Comment (thomas-robinson) [inline full/full_coupler_mod.F90:1711]: this also calls the clocks for flux_ocean_ice

File: full/full_coupler_mod.F90 [hunk 3]
```diff
@@ -1694,19 +1693,80 @@ subroutine coupler_flux_check_stocks(nc, Time, Atm, Land, Ice, Ocean_state, coup
...
+    call fms_mpp_clock_begin(coupler_clocks%flux_ocean_to_ice)
+
+    !Ice intent is In, used only for accessing Ice%area and knowing if we are on an Ice pe
+    call flux_ocean_to_ice(Ocean, Ice, Ocean_ice_boundary)
+
+    call fms_mpp_clock_end(coupler_clocks%flux_ocean_to_ice)
+
+  end subroutine coupler_flux_ocean_to_ice
+
+!> \brief This subroutine calls flux_ocean_to_ice
```
Comment (thomas-robinson) [inline full/full_coupler_mod.F90:1735]: This is also too brief.  This routine sets clocks and does a check before it calls the routine

**Review by thomas-robinson — CHANGES_REQUESTED** [review]

File: full/full_coupler_mod.F90 (hunk 2 above)
Comment (mlee03) [reply full/full_coupler_mod.F90]: Added.

File: full/full_coupler_mod.F90 (hunk 3 above)
Comment (mlee03) [reply full/full_coupler_mod.F90]: Added.

**Review by thomas-robinson — APPROVED** [review]

**Pull request merged**